codegen/src/main/java/org/web3j/codegen/TruffleJsonFunctionWrapperGenerator.java

```diff
@@ -50,66 +54,107 @@
+    public static final String COMMAND_TRUFFLE = "truffle";
+    public static final String COMMAND_GENERATE = "generate";
+    public static final String COMMAND_PREFIX = COMMAND_TRUFFLE + " " + COMMAND_GENERATE;
 
-    private String jsonFileLocation;
+    private final String jsonFileLocation;
+    private final boolean generateBothCallAndSend;
 
     public TruffleJsonFunctionWrapperGenerator(
             String jsonFileLocation,
             String destinationDirLocation,
             String basePackageName,
-            boolean useJavaNativeTypes) {
+            boolean useJavaNativeTypes,
+            boolean generateBothCallAndSend) {
 
         super(new File(destinationDirLocation), basePackageName, useJavaNativeTypes);
         this.jsonFileLocation = jsonFileLocation;
+        this.generateBothCallAndSend = generateBothCallAndSend;
     }
 
-    public static void run(String[] args) throws Exception {
-        if (args.length < 1 || !"generate".equals(args[0])) {
-            exitError(USAGE);
-        } else {
-            main(tail(args));
+    public static void main(String[] args) {
+        if (args.length > 0 && args[0].equals(COMMAND_TRUFFLE)) {
+            args = tail(args);
         }
-    }
-
-    public static void main(String[] args) throws Exception {
-
-        String[] fullArgs;
-        if (args.length == 5) {
-            fullArgs = new String[args.length + 1];
-            fullArgs[0] = JAVA_TYPES_ARG;
-            System.arraycopy(args, 0, fullArgs, 1, args.length);
-        } else {
-            fullArgs = args;
+        if (args.length > 0 && args[0].equals(COMMAND_GENERATE)) {
+            args = tail(args);
         }
+        CommandLine cmd = new CommandLine(new PicocliRunner());
+        cmd.parseWithHandlers(
+                new RunLast().useOut(System.out).useAnsi(Help.Ansi.AUTO),
+                RethrowExceptionHandler.HANDLER,
+                args);
+    }
 
-        if (fullArgs.length != 6) {
-            exitError(USAGE);
+    @Command(
+            name = COMMAND_PREFIX,
+            mixinStandardHelpOptions = true,
+            version = "4.0",
+            sortOptions = false)
+    private static class PicocliRunner implements Callable<Void> {
+
+        @Option(
+                names = {"-jt", JAVA_TYPES_ARG},
+                description = "use native Java types.",
+                showDefaultValue = ALWAYS)
+        boolean useJavaNativeTypes;
+
+        @Option(
+                names = {"-st", SOLIDITY_TYPES_ARG},
+                description = "use solidity types.")
+        boolean useSolidityTypes;
+
+        @Parameters(index = "0")
+        String jsonFileLocation;
+
+        @Option(
+                names = {"-o", "--outputDir"},
+                description = "destination base directory.",
+                required = true)
+        private String destinationDirLocation;
+
+        @Option(
+                names = {"-B", "--generateBoth"},
```

Comment:
rename this to generateCallAndSend with "-cs"

Reply:
I think it will be better to be consistent with `SolidityFunctionWrapperGenerator` which use `"-B", "--generateBoth"` ?

https://github.com/web3j/web3j/blob/master/codegen/src/main/java/org/web3j/codegen/SolidityFunctionWrapperGenerator.java#L263